File: sdkwork-ai-sdk-typescript/src/types/message.ts
import type { ToolCall } from './tool-call';

/** 消息 */
export interface Message {
  /** 角色 */
  role?: string;
  /** 消息内容 */
  content?: unknown;
  /** 拒绝消息 */
  refusal?: string;
  /** 注解列表 */
  annotations?: unknown[];
  /** 思维链内容 */
  reasoning_content?: string;
  /** 工具调用列表 */
  tool_calls?: ToolCall[];
}

File: sdkwork-ai-sdk-java/src/main/java/com/sdkwork/ai/model/Message.java
package com.sdkwork.ai.model;

public class Message {
    private String role;
    private Object content;
    private String refusal;
    private List<Object> annotations;
    private String reasoningContent;
    private List<ToolCall> toolCalls;

    public String getRole() {
        return this.role;
    }
    
    public void setRole(String role) {
        this.role = role;
    }

    public Object getContent() {
        return this.content;
    }
    
    public void setContent(Object content) {
        this.content = content;
    }

    public String getRefusal() {
        return this.refusal;
    }
    
    public void setRefusal(String refusal) {
        this.refusal = refusal;
    }

    public List<Object> getAnnotations() {
        return this.annotations;
    }
    
    public void setAnnotations(List<Object> annotations) {
        this.annotations = annotations;
    }

    public String getReasoningContent() {
        return this.reasoningContent;
    }
    
    public void setReasoningContent(String reasoningContent) {
        this.reasoningContent = reasoningContent;
    }

    public List<ToolCall> getToolCalls() {
        return this.toolCalls;
    }
    
    public void setToolCalls(List<ToolCall> toolCalls) {
        this.toolCalls = toolCalls;
    }
}
